### PR TITLE
[BugFix] Fix write fail error message

### LINKED ISSUE #25943: write timeout when insert invalid data
### Steps to reproduce the behavior (Required)
```
CREATE TABLE test_semi_struct_err(
    v1 INT,
    map_int_float MAP<INT, FLOAT>,
    s struct<s1 int, s2 char(5)>
)ENGINE=OLAP
DUPLICATE KEY(v1)
COMMENT "OLAP"
DISTRIBUTED BY HASH(v1) BUCKETS 3
PROPERTIES (
    "in_memory"="false",
    "enable_persistent_index"="false",
    "replicated_storage"="true",
    "compression"="LZ4"
);

insert into test_semi_struct_err values(1, map{1: 1.1}, row(1, "123456"));
```

### Expected behavior (Required)
fail with message

### Real behavior (Required)
fail with timeout

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`

### PR DESCRIPTION
**Pull request by meegoo**
Fixes #25943

## What type of PR is this:
- [x] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.1
  - [x] 3.0
  - [x] 2.5
  - [ ] 2.4

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/runtime/lake_tablets_channel.cpp [hunk 1]
```diff
@@ -75,7 +75,7 @@ class LakeTabletsChannel : public TabletsChannel {
 
     void abort() override;
 
-    void abort(const std::vector<int64_t>& tablet_ids) override { return abort(); }
+    void abort(const std::vector<int64_t>& tablet_ids, const std::string& reason) override { return abort(); }
```
Comment (imay) [inline be/src/runtime/lake_tablets_channel.cpp:78]: Why not use Status as the root cause?

Comment (meegoo) [reply be/src/runtime/lake_tablets_channel.cpp:78]: this function called by remote node through rpc, It is not only used for abort when there is an exception, but also for partitions whose data is empty which abort is normal.

File: be/src/runtime/local_tablets_channel.cpp [hunk 2]
```diff
@@ -265,6 +268,10 @@ void LocalTabletsChannel::add_chunk(Chunk* chunk, const PTabletWriterAddChunkReq
     // This will only block the bthread, will not block the pthread
     count_down_latch.wait();
 
+    // Abort tablets which primary replica already failed
+    _abort_replica_tablets(request, response->status().status_code() == 0 ? "" : response->status().error_msgs()[0],
```
Comment (chaoyli) [inline be/src/runtime/local_tablets_channel.cpp:272]: Why to call `_abort_replica_tablets` when `response->status().status_code() == 0`?

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.1

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-3.0

**Comment by wanpengfei-git** [comment]
@Mergifyio backport branch-2.5

**Comment by meegoo** [comment]
@mergify backport branch-2.5

**Pull request merged**